src/parser.ts

```diff
@@ -4445,13 +4457,105 @@ export function parseImportExpression(
   if (parser.getToken() === Token.Ellipsis) report(parser, Errors.InvalidSpreadInImport);
 
   const source = parseExpression(parser, context, 1, inGroup, parser.tokenPos, parser.linePos, parser.colPos);
+  const node: ESTree.ImportExpression = {
+    type: 'ImportExpression',
+    source
+  };
+
+  if (context & Context.OptionsNext) {
+    let options: ESTree.Expression | null = null;
+
+    if (parser.getToken() === Token.Comma) {
+      consume(parser, context, Token.Comma);
+
+      while (parser.getToken() !== Token.RightParen && parser.getToken() !== Token.Comma) {
+        if (parser.getToken() === Token.InKeyword) {
+          consume(parser, context, Token.InKeyword);
+        }
+        options = parseExpression(parser, context, 1, inGroup, parser.tokenPos, parser.linePos, parser.colPos);
```

Comment:
I'm not sure about this, they pass the tests, but I was really just looking for “no crash”, but I think if I add tests of the expected AST to test262/test/language/expressions/dynamic-import/import-attributes/2nd-param-in.js it will crash. Tomorrow I will continue and add more tests from test262 with the AST after parsing.

Reply:
Yes, this is not right.
"in" is handled by parseBinaryExpression. The reason that parseExpression didn't cover it, is probably Context.DisallowIn is turned on. Probably need just `(context | Context.DisallowIn) ^ Context.DisallowIn` to turn it off.

Reply:
Okay perfect, it looks better now. I've added the failing test with the AST and corrected it with what you told me.